src/deposit_monitor.rs

```diff
@@ -86,9 +145,13 @@ impl DepositMonitor {
 
     /// Check pending deposits confirmed to the monitored addresses
     pub fn get_pending_deposits(
-        &self,
+        &mut self,
         chain_tip: &BlockRef,
     ) -> Result<Vec<CreateDepositRequestBody>, Error> {
+        tracing::debug!(
+            "Checking for pending deposits at chain tip: {}",
+            chain_tip.block_height
```

Comment:
nit: move `block_height` to a field, for easier parsing when using structured logs. Also at the beginning of `get_deposit_from_utxo`

Reply:
whoops, accidentally left debug logs, [cb6d30f](https://github.com/BitcoinL2-Labs/sPoX/pull/8/commits/cb6d30fceaa37cabc39323849f699f7fb686c9f8), [d68606b](https://github.com/BitcoinL2-Labs/sPoX/pull/8/commits/d68606bc7b2fabd4bcd695cc9eb90f1c9bf24024)

Reply:
yea, deleted it [4380bae](https://github.com/BitcoinL2-Labs/sPoX/pull/8/commits/4380bae3e437f8bce56d739376c38a4a19c3bf5c)